Fix image TV display via hash

tanzirev opened the pull request:
Если выбранные изображения имеют одинаковые названия, но при этом это разные изображения, то превью таких изображений не обновляются на разных страницах, т.к. браузер кеширует прошлое изображение. 
(If the selected images have the same name, but they are different images, then the previews of such images are not updated on different pages, because the browser caches the previous image.)

tanzirev committed:
FIX update preview

Если выбранные изображения имеют одинаковые названия, но при этом это разные изображения, то превью таких изображений не обновляются на разных страницах, т.к. браузер кеширует прошлое изображение. 
(If the selected images have the same name, but they are different images, then the previews of such images are not updated on different pages, because the browser caches the previous image.)

tanzirev commented:
@cla-bot check

Jako reviewed (changes requested):
If I understand the issue correctly, the preview for images with the same path and source is the same on different MODX sites.

I suggest using a hash of the MODX site_id instead of a random string. Then the thumbnail can be cached for the same site. Is this suitable?

Ruslan-Aleev commented:
As I understand it, here is about the fact that the cache of the old image is displayed in the manager when replacing the image with a new one, but with the same name (or just images with the same names). Those MODX site_id will not help with this issue.

Jako committed:
Switch to a file based hash

Jako committed:
Switch to a file based hash

Jako commented:
I have changed the code to use a file (size + modification time) based hash value (if the file exist). Then the browser can cache the results.

Jako committed:
Fix phpThumb private parameter issue

Fix warning: Detected attempt of using private parameter `version` (for internal usage) of phpThumb that not allowed and insecure
